scripts/west_commands/bindesc.py

```diff
@@ -0,0 +1,284 @@
+# Copyright (c) 2023 Yonatan Schachter
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from textwrap import dedent
+import struct
+
+from west.commands import WestCommand
+from west import log
+
+from elftools.elf.elffile import ELFFile
+from intelhex import IntelHex
+
+# Copied from scripts/build/uf2conv.py
+UF2_MAGIC_START0 = 0x0A324655 # 'UF2\n'
+UF2_MAGIC_START1 = 0x9E5D5157 # Randomly selected
+
+
+# Copied from scripts/build/uf2conv.py
+def convert_from_uf2(buf):
+    global appstartaddr
+    numblocks = len(buf) // 512
+    curraddr = None
+    outp = []
+    for blockno in range(numblocks):
+        ptr = blockno * 512
+        block = buf[ptr:ptr + 512]
+        hd = struct.unpack(b'<IIIIIIII', block[0:32])
+        if hd[0] != UF2_MAGIC_START0 or hd[1] != UF2_MAGIC_START1:
+            print('Skipping block at ' + ptr + '; bad magic')
+            continue
+        if hd[2] & 1:
+            # NO-flash flag set; skip block
+            continue
+        datalen = hd[4]
+        if datalen > 476:
+            assert False, 'Invalid UF2 data size at ' + ptr
+        newaddr = hd[3]
+        if curraddr is None:
+            appstartaddr = newaddr
+            curraddr = newaddr
+        padding = newaddr - curraddr
+        if padding < 0:
+            assert False, 'Block out of order at ' + ptr
+        if padding > 10*1024*1024:
+            assert False, 'More than 10M of padding needed at ' + ptr
+        if padding % 4 != 0:
+            assert False, 'Non-word padding size at ' + ptr
+        while padding > 0:
+            padding -= 4
+            outp += b'\x00\x00\x00\x00'
+        outp.append(block[32 : 32 + datalen])
+        curraddr = newaddr + datalen
+    return b''.join(outp)
+
+
+class Bindesc(WestCommand):
+    EXTENSIONS = ['bin', 'hex', 'elf', 'uf2']
+    MAGIC = 0x2c31b9b9
+    DESCRIPTORS_END = 0xffffffff
+
+    TYPE_UINT = 0
+    TYPE_STR = 1
```

Comment:
These and the corresponding definitions in the C header should be commented to say "if you change this, you need to keep $THE_OTHER_ONE in sync". Similar comment for any other shared constants.